module-1.md

```diff
@@ -0,0 +1,73 @@
+# 1: What is mesh?
+
+Total length:  1.5 hour
+Planned:      1.25 hour
+
+| **Section**                                        | **Length** |
+|----------------------------------------------------|------------|
+| **1: Topic introduction**                          | 15 min     |
+| **2: Hands-on and material exploration**           | 40 min     |
+| **3: Discussion and reflection activity**          | 20 min     |
+| **4: Wrap-up and week ahead**                      | 5 min      |
+
+## 1: Topic introduction
+
+- Introduce yourself, why you are interested in this workshop
+- Discuss familiar topics like net neutrality, internet service providers, and internet censorship
+- Discuss social, political and technical reasons for a mesh
+- Succinct high-level slides on many pieces that will be explored in more detail in upcoming modules:
+    - two minutes on historic Toronto networks
+    - two minutes on community mesh networks in North America and other parts of the world
+    - general landscape of technologies like flashing routers with OpenWrt to install some mesh routing software
+- Set realistic expectations of the state of mesh technology
+
+## 2: Hands-on and material exploration
+
+### Objectives
+
+- Experience local peer-to-peer applications that are relatable to those we are familiar with from the Internet (e.g. Facebook, Dropbox, Whatsapp, Twitter, Spotify)
+- "Run your own app" on pre-configured nodes and contrast "serverless" applications on a distributed physical network vs. traditional centralized edge services via ISP networks
+
+**Ideas for peer-to-peer applications:**
+
+_Preference to those that require minimal steps (e.g. exchanging keys, after `docker start`) and have a familiar UI._
+
+    - Secure Scuttlebutt / Patchwork (need to pin public keys)
+    - IPFS (auto peer but need to share content hash)
+    - RetroShare
+    - (Chat?)
+    - (Collaborative text editor?)
+    - (Audio/Video streaming, could be part of IPFS?)
+    - See [Portable Network Kit](http://pnkgo.com) for types of services they run on a Raspberry Pi
+
+### Materials
+
+A _Raspberry Pi 3 Kit_ for each student that includes:
+
+- USB WiFi radio that is ad-hoc or 802.11s-capable (e.g. TP-Link TL-WN722N)
+- SD card with [mesh-orange](https://github.com/tomeshnet/mesh-orange) pre-configured to mesh wirelessly with cjdns
+- Tools to discover node identity in the ad-hoc network and share human-readable names with others (e.g. [ssb-dns](https://github.com/ansuz/dnssb))
+- Contain Docker image for Secure Scuttlebutt and multiple Patchwork clients (i.e. [scuttlebot](https://github.com/ssbc/scuttlebot), [patchwork](https://github.com/ssbc/patchwork), [patchbay](https://github.com/ssbc/patchbay), [patchfoo](https://github.com/ssbc/patchfoo))
```

Comment:
@ansuz Do you know if patchwork and patchbay can be installed on a Raspberry Pi and accessed over browser from a laptop client?

Reply:
Patchwork is an electron app, and I don't think there is a way to access it remotely except by X11 Forwarding.

patchbay, patchfoo, and several other clients are written to run in the browser, so they'd just need a means of restricting access to maybe a single cjdns ip.

Initial sync with ssb can be a bit heavy for the CPU, so I'd recommend that nobody start with an existing ssb feed.

Reply:
[scuttlebot](https://github.com/ssbc/scuttlebot) can be installed on an raspberry-pi, and it's all you need to access the network at a basic level. A basic UI could be made targeting this platform for whatever display you have. In my case, I plan to show basic (mainly connectivity) details of the ssb nodes attached to my mesh network on the installed sensehat. More details on getting started with sbot: https://www.scuttlebutt.nz/guides/cli-first-steps.html

Reply:
@vuldin Do you know if Patchbay can be run on the Pi as well, and then exposed on some port and accessed through browser of a NAT client? The command line is cool, but since this is the first workshop for a general audience, where cli is usually unfamiliar, it's best to start at the browser and then cli to show _this is the actual message that is exchanged_.

Reply:
@benhylau I'm not sure about patchbay since I haven't tried it. My plan is to just create a websocket wrapper for sbot that allows a webapp to communicate with the service. Then I can have that webapp run on any other more graphically capable device such as my phone or laptop.

It would be cool to have the Pi connect to the mesh and then provide that connection to my phone (along with connection to the secure-scuttlebutt network). This connection could be done either with the built-in Pi wifi (if available) or bluetooth. All this along with a battery-powered Pi means an entirely mobile mesh-connected system with ssb (and any other P2P app/service you run on the Pi).

Reply:
Are you also building the webapp UI that talks to the ssb websocket? RE: mesh connection, yes the Raspberry Pi 3 configures the on-board WiFi as hostap and all ipv6 NAT traffic is passed to the cjdns tun interface: https://github.com/tomeshnet/prototype-cjdns-pi/blob/master/scripts/hostapd/nat.sh#L11

Reply:
@benhylau I'm working on the UI, yes. I'm only at the point of having the framework in place, and it has basic connectivity/communication with the server. Once I have sbot sending data to websocket channels that are successfully displaying on the client I'll push the initial commit of the web project. Keep in mind that that level of functionality will be far from being a patchwork alternative, but it will be a good start.

Reply:
By the way @benhylau , that's awesome that cjdns can be the provider for the IPv6 address for mesh networking, and that you are still also able to connect to the broader network provided by cjdns (when connected to an access point). I will still need to do testing since I'm unfamiliar with how this all works... I'll share any notes.